src/Views/AppInfoView.vala

```diff
@@ -212,37 +211,38 @@ namespace AppCenter.Views {
 
             var header_grid = new Gtk.Grid ();
             header_grid.column_spacing = 12;
+            header_grid.row_spacing = 6;
             header_grid.halign = Gtk.Align.CENTER;
             header_grid.margin = 12;
-            header_grid.width_request = 800;
+            header_grid.width_request = 900; /* Must wide enought to fit long package name and progress bar */
+            header_grid.hexpand = true;
             header_grid.attach (image, 0, 0, 1, 2);
             header_grid.attach (package_name, 1, 0, 1, 1);
+
             if (!package.is_os_updates) {
-                header_grid.attach (package_author, 1, 1, 3, 1);
+                header_grid.attach (package_author, 1, 1, 1, 1);
                 header_grid.attach (app_version, 2, 0, 1, 1);
             } else {
                 package_summary.get_style_context ().add_class (Gtk.STYLE_CLASS_DIM_LABEL);
-                package_summary.valign = Gtk.Align.START;
-                header_grid.attach (package_summary, 1, 1, 3, 1);
+                package_summary.valign = Gtk.Align.CENTER;
+                header_grid.attach (package_summary, 1, 1, 2, 1);
             }
 
-            if (!package.is_local) {
-                var download_button_grid = new Gtk.Grid ();
-                download_button_grid.orientation = Gtk.Orientation.VERTICAL;
-                download_button_grid.valign = Gtk.Align.END;
-                download_button_grid.halign = Gtk.Align.END;
-                download_button_grid.row_spacing = 3;
+            action_stack.valign = Gtk.Align.CENTER;
+            action_stack.halign = Gtk.Align.END;
+            action_stack.hexpand = true;
+
+            header_grid.attach (action_stack, 3, 0, 1, 1);
 
+            if (!package.is_local) {
                 app_download_size_cancellable = new Cancellable ();
 
                 app_download_size_label = new Gtk.Label (null);
                 app_download_size_label.visible = false;
                 app_download_size_label.halign = Gtk.Align.END;
+                app_download_size_label.valign = Gtk.Align.CENTER;
```

Comment:
I feel this label should be attached directly below the install button, because it's related to it.

Reply:
I  think there was a technical problem with that but I'll try it.